### ISSUE TITLE
add_index migration with to_tsvector fails

### ISSUE THREAD

**Issue by mmb** [rails/rails#25270]
### Steps to reproduce

Create a migration like this and run it against a PostgreSQL database:

```
add_index :documents, "to_tsvector('english', name)", using: :gin, name: 'documents_idx'
```

This doc suggests that it will work:

http://edgeguides.rubyonrails.org/active_record_postgresql.html#full-text-search

### Expected behavior

It should create the index.

### Actual behavior
```
ERROR:  column "to_tsvector('english', name)" does not exist
STATEMENT:  CREATE  INDEX  "src_image_name_tsvector" ON "src_images" USING gin ("to_tsvector('english', name)")
```

### System configuration
**Rails version**: 4.2.6

**Ruby version**: 2.3.0

If it's not possible, the doc should be changed.

**Comment by kamipo** [comment]
Looks like that this issue was fixed at #23393. Could you try 5.0.0.rc1?

**Comment by vipulnsward** [comment]
Works as expected on master, and rc1 as @kamipo pointed out
- https://gist.github.com/vipulnsward/d26e33ee55909945a07ed5096cce2d77
